src/editor.cpp

```diff
@@ -0,0 +1,580 @@
+        char kwNameStart = *iterator;
+        std::advance(iterator, Length::KW_NAME);
+
+        // if keyword starts with #
+        if (constants::POUND_KW == kwNameStart)
+        {
+            // Note existing keyword data length
+            dataLength = get2BytesValue(iterator);
+
+            // Jump past 2Byte keyword length + data
+            std::advance(iterator, sizeof(types::PoundKwSize));
+        }
+        else
+        {
+            // Note existing keyword data length
+            dataLength = *iterator;
+
+            // Jump past keyword length and data
+            std::advance(iterator, sizeof(KwSize));
+        }
+
+        if (m_thisRecord.keywordName == thisKwName)
+        {
+            m_thisRecord.kwdDataOffset =
+                std::distance(m_vpdBytes.cbegin(), iterator);
+            m_thisRecord.kwdDataLength = dataLength;
+        std::cout<<"DBG:return getTheKwdOffsetAndLength \n";
+	return;
+        }
+
+        // jump the data of current kwd to point to next kwd name
+        std::advance(iterator, dataLength);
+    }
+
+    throw DataException("Keyword not found");
+}
+
+void vpdEditor::writeNewDataToVPD()
+{
+	std::cout<<"DBG:Eneter writeNewDataToVPD \n";
+    auto dataLength = m_thisRecord.newData.size() <= m_thisRecord.kwdDataLength
+                          ? m_thisRecord.newData.size()
+                          : m_thisRecord.kwdDataLength;
+    auto dataStart = m_thisRecord.newData.cbegin();
+    auto dataEnd = dataStart;
+    std::advance(dataEnd, dataLength);
+
+    auto iteratoreToWriteToVpd = m_vpdBytes.begin();
+    std::advance(iteratoreToWriteToVpd, m_thisRecord.kwdDataOffset);
+
+    std::copy(dataStart, dataEnd, iteratoreToWriteToVpd);
+
+    m_vpdFileStream.seekp(m_vpdStartOffset + m_thisRecord.kwdDataOffset,
+                          std::ios::beg);
+
+    dataStart = m_thisRecord.newData.cbegin();
+    dataEnd = dataStart;
+    std::advance(dataEnd, dataLength);
+    std::copy(dataStart, dataEnd,
+              std::ostreambuf_iterator<char>(m_vpdFileStream));
+std::cout<<"DBG:return writeNewDataToVPD \n";
+}
+
+void vpdEditor::updateRecordECC()
+{
+	std::cout<<"DBG:Eneter updateRecordECC \n";
+    auto itrToRecordData = m_vpdBytes.cbegin();
+    std::advance(itrToRecordData, m_thisRecord.recordOffset);
+
+    auto itrToRecordECC = m_vpdBytes.cbegin();
+    std::advance(itrToRecordECC, m_thisRecord.recordECCoffset);
+
+    auto l_status = vpdecc_create_ecc(const_cast<uint8_t*>(&itrToRecordData[0]),
+                                      m_thisRecord.recordDataLength,
+                                      const_cast<uint8_t*>(&itrToRecordECC[0]),
+                                      &m_thisRecord.recordECCLength);
+    if (l_status != VPD_ECC_OK)
+    {
+        throw EccException("Ecc update failed");
+    }
+
+    auto end = itrToRecordECC;
+    std::advance(end, m_thisRecord.recordECCLength);
+
+    m_vpdFileStream.seekp(m_vpdStartOffset + m_thisRecord.recordECCoffset,
```

Comment:
File exceptions handling can be done in places required. Also, errors if needed.